### ISSUE TITLE
[REQ] Package as flatpak

### ISSUE THREAD

**Issue by 1989gironimo** [LedgerHQ/ledger-live-desktop#2116]
- [x] I have checked this feature was not yet requested.
analogue to this request: https://github.com/LedgerHQ/ledger-live-desktop/issues/1431

There are several devices running ordinary GNU/Linux (on ARM) coming out this year (Librem 5, Pinephone), which tends to be more secure and suitable for managing the Ledger devices. One of it, the Librem 5, uses Flatpak for distributing apps via their upcoming PureOS store. It seems a little bit niche and I don't know how much work it is to distribute another package format (and asking for UI enhancements as well, because the screen doesn't have desktop proportions) but I think it does have its usecase.

**Comment by acpuchades** [comment]
I'd love to see this happening!Is there any way we can help?

**Comment by gre** [comment]
We were currently looking at supporting .snap and it is challenging regarding the usb support.
flatpak is something we can take a look too and maybe a less challenging one. cc @IAmMorrow @MortalKastor

**Comment by MortalKastor** [comment]
Both Librem 5 and Pinephone are an interesting take on smartphones and look like pretty cool devices all around, but adding support would be far from an easy task:
- we currently don't support ARM. We would need all our native dependencies to do too, and I don't know the status of that.
- the framework we rely on to bundle our app doesn't offer flatpak out of the box. It might not be difficult to make this ourselves, but for them to not offer it may be an indication that it's not an easy feat.
- the app would be unusable on a 5" phone screen, unless we port (rewrite) our React Native mobile UI to our Electron codebase.
- you probably don't want to run an Electron app on your phone. The trade-off offered by Electron is to have a single codebase running in the exact same way on all the OSes at the cost of a bigger CPU and RAM footprint. It's a compelling offer on a computer, but doesn't make sense on a ressource constrained device like a phone.

That being said, both flatpak and arm support are interesting on their own and worth looking into, if not for Linux on mobile.

**Comment by tjanez** [comment]
> Both Librem 5 and Pinephone are an interesting take on smartphones and look like pretty cool devices all around, but adding support would be far from an easy task:

Agreed. Perhaps it would be better to open a new issue to track that since support for ARM and support for Flatpak are two independent things?

I would love to use Ledger Live in a Flatpak sandbox on my x86_64 workstation.

> the framework we rely on to bundle our app doesn't offer flatpak out of the box. It might not be difficult to make this ourselves, but for them to not offer it may be an indication that it's not an easy feat.

I don't know what you use, but Electron Builder doesn't support Flatpaks yet, even though there has been a lot of demand: https://github.com/electron-userland/electron-builder/issues/512.

Regarding Snap vs. Flatpak, the latter has been gaining quite a lot of "buzz" [according to Google Trends](https://trends.google.com/trends/explore?q=snapcraft,flatpack).

**Comment by acpuchades** [comment]
Any update on this? If not already working on it, is there any way we can help?

**Comment by anononez** [comment]
I would love to see this feature. +1

**Comment by tjanez** [comment]
> 
> > the framework we rely on to bundle our app doesn't offer flatpak out of the box. It might not be difficult to make this ourselves, but for them to not offer it may be an indication that it's not an easy feat.
> 
> I don't know what you use, but Electron Builder doesn't support Flatpaks yet, even though there has been a lot of demand: [electron-userland/electron-builder#512](https://github.com/electron-userland/electron-builder/issues/512).
 
Update: On Apr 9, 2021, experimental support for building Flatpaks was merged into Electron Builder: electron-userland/electron-builder#5711.

**Comment by mispp** [comment]
is flatpak version going to be worked on then?
appiamge seems to be there, but snap or flatpak for autoupdate would be great

**Comment by tidux** [comment]
Flatpak in particular would be useful for integration with secure, read-only-root distributions such as Fedora Silverblue, or for non-glibc distributions like Alpine.
